### PR TITLE
🐛 Fix aws checksum issue

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
Add the sha256 checksum of an S3 object directly in the `Metadata` field instead of adding it via the `ChecksumSHA256` field. 
The reason why this is required is discussed in some detail in the issue listed below. 

## Related issue/s
- fixes https://github.com/ITISFoundation/osparc-simcore/issues/4885

## How to test
This was tested manually using an S3 test bucket on my local deployment. I believe all multipart uploads via storage to AWS S3 are currently broken by this, hence the multipart upload in the client e2e tests should actually test this. Hence, instead of introducing additional tests I think it is important, that I propagate the multipart upload in the public API to production, so that those tests will also run against AWS S3.

### CODE REVIEW

File: services/storage/src/simcore_service_storage/s3_client.py [hunk 1]
```diff
@@ -170,11 +170,15 @@ async def create_multipart_upload_links(
         file_id: SimcoreS3FileID,
         file_size: ByteSize,
         expiration_secs: int,
+        sha256_checksum: SHA256Str | None,
     ) -> MultiPartUploadLinks:
         # NOTE: ensure the bucket/object exists, this will raise if not
         await self.client.head_bucket(Bucket=bucket)
         # first initiate the multipart upload
-        response = await self.client.create_multipart_upload(Bucket=bucket, Key=file_id)
+        create_input: dict[str, Any] = dict(Bucket=bucket, Key=file_id)
```
Comment (GitHK) [inline services/storage/src/simcore_service_storage/s3_client.py:178]:
```suggestion
        create_input: dict[str, Any] = {"Bucket": bucket, "Key": file_id}
```

File: services/storage/src/simcore_service_storage/s3_client.py [hunk 2]
```diff
@@ -170,11 +170,15 @@ async def create_multipart_upload_links(
...
+        if sha256_checksum:
+            create_input.update(Metadata={"sha256_checksum": sha256_checksum})
```
Comment (GitHK) [inline services/storage/src/simcore_service_storage/s3_client.py:180]:
```suggestion
            create_input["Metadata"] = {"sha256_checksum": sha256_checksum}
```

**Review by GitHK — APPROVED** [review]
Very good. Thanks for taking care of this.

Just two minor suggestions.

**Comment by bisgaard-itis** [comment]
> looks good. so if I got that right, you now keep the checksum in the metadata instead of using the S3 built-ins right? same as what we observed RClone is doing?

Yes, exactly

**Pull request merged**